2008-10-23  Aurelien Jarno  <aurelien@aurel32.net>

        * sunrpc/rpc_main.c(mkfile_output): Properly handle filename 
	without dot.  Properly terminate the string with a null byte.

diff a/sunrpc/rpc_main.c b/sunrpc/rpc_main.c
--- a/sunrpc/rpc_main.c
+++ b/sunrpc/rpc_main.c
@@ -995,7 +995,13 @@ mkfile_output (struct commandline *cmd)
 	abort ();
       temp = rindex (cmd->infile, '.');
       cp = stpcpy (mkfilename, "Makefile.");
-      strncpy (cp, cmd->infile, (temp - cmd->infile));
+      if (temp)
+	{
+	  strncpy (cp, cmd->infile, (temp - cmd->infile));
+	  cp[temp - cmd->infile] = '\0';
+	}
+      else
+	strcpy (cp, cmd->infile);
     }
   else
     mkfilename = (char *) cmd->outfile;
